music_assistant/providers/resonate/player.py

```diff
@@ -0,0 +1,152 @@
+"""Resonate Player implementation."""
+
+from __future__ import annotations
+
+import time
+from collections.abc import Callable
+from typing import TYPE_CHECKING, cast, override
+
+from aioresonate.server import AudioFormat as ResonateAudioFormat
+from aioresonate.server import PlayerEvent, VolumeChangedEvent
+from music_assistant_models.constants import PLAYER_CONTROL_NONE
+from music_assistant_models.enums import ContentType, PlaybackState, PlayerFeature, PlayerType
+from music_assistant_models.media_items import AudioFormat
+from music_assistant_models.player import DeviceInfo
+
+from music_assistant.models.player import Player, PlayerMedia
+
+if TYPE_CHECKING:
+    from aioresonate.server import Player as PlayerInstance
+
+    from .provider import ResonateProvider
+
+
+class ResonatePlayer(Player):
+    """A resonate audio player in Music Assistant."""
+
+    player: PlayerInstance
+    unsub_event_cb: Callable[[], None]
+
+    def __init__(self, provider: ResonateProvider, player_id: str) -> None:
+        """Initialize the Player."""
+        super().__init__(provider, player_id)
+        player = provider.server.get_player(player_id)
+        assert player is not None
+        self.player = player
```

Comment:
Not a fan of this player.player - I assume this is the Resonate player object ?
So should we maybe call it "api" or "resonate" or "resonate_player" here ?